test/regress/run_regression.py

```diff
@@ -11,36 +11,330 @@
+# benchmark and `run()` runs the actual test. Most testers can invoke the
+# `run()` method in the base class, which calls the cvc5 binary with a set of
+# arguments and checks the expected output (both stdout and stderr) as well as
+# the exit status.
+#
+# To add a new tester, add a class and add it to the `g_tester` dictionary.
+################################################################################
+
+
+class BaseTester(Tester):
+    def __init__(self):
+        pass
+
+    def run(self, benchmark_info):
+        return super().run(benchmark_info)
+
+
+class UnsatCoreTester(Tester):
+    def __init__(self):
+        pass
+
+    def applies(self, benchmark_info):
+        return (
+            benchmark_info.benchmark_ext != ".sy"
+            and (
+                "unsat" in benchmark_info.expected_output.split()
+                or "entailed" in benchmark_info.expected_output.split()
+            )
+            and "--no-produce-unsat-cores" not in benchmark_info.command_line_args
+            and "--no-check-unsat-cores" not in benchmark_info.command_line_args
+            and "--check-unsat-cores" not in benchmark_info.command_line_args
+            and "sygus-inference" not in benchmark_info.benchmark_content
+            and "--unconstrained-simp" not in benchmark_info.command_line_args
+        )
 
-SCRUBBER = 'SCRUBBER:'
-ERROR_SCRUBBER = 'ERROR-SCRUBBER:'
-EXPECT = 'EXPECT:'
-EXPECT_ERROR = 'EXPECT-ERROR:'
-EXIT = 'EXIT:'
-COMMAND_LINE = 'COMMAND-LINE:'
-REQUIRES = 'REQUIRES:'
+    def run(self, benchmark_info):
+        return super().run(
+            benchmark_info._replace(
+                command_line_args=benchmark_info.command_line_args
+                + ["--check-unsat-cores"]
+            )
+        )
+
+
+class ProofTester(Tester):
+    def __init__(self):
+        pass
+
+    def applies(self, benchmark_info):
+        expected_output_lines = benchmark_info.expected_output.split()
+        return (
+            benchmark_info.benchmark_ext != ".sy"
+            and (
+                "unsat" in benchmark_info.expected_output.split()
+                or "entailed" in benchmark_info.expected_output.split()
+            )
+            and "--no-produce-proofs" not in benchmark_info.command_line_args
+            and "--no-check-proofs" not in benchmark_info.command_line_args
+            and "--check-proofs" not in benchmark_info.command_line_args
+        )
+
+    def run(self, benchmark_info):
+        return super().run(
+            benchmark_info._replace(
+                command_line_args=benchmark_info.command_line_args + ["--check-proofs"]
+            )
+        )
+
+
+class ModelTester(Tester):
+    def __init__(self):
+        pass
+
+    def applies(self, benchmark_info):
+        expected_output_lines = benchmark_info.expected_output.split()
+        return (
+            ("sat" in expected_output_lines or "unknown" in expected_output_lines)
```

Comment:
I expected to see `benchmark_info.benchmark_ext != ".sy"` here and in `AbductTester.applies`. Does it make sense to run `ModelTester` and `AbductTester` on sygus benchmarks?

Reply:
Yeah, that's a good point. I've used the checks of the original code but I agree that it would be better to not run those on SyGuS benchmarks.